homeassistant/components/flux_led/config_flow.py

```diff
@@ -0,0 +1,234 @@
+        return await self.async_step_prompt_options()
+
+    async def async_step_prompt_options(self, user_input=None):
+        """Manage the options."""
+
+        errors = {}
+
+        if user_input is not None:
+            self._global_options = {
+                CONF_AUTOMATIC_ADD: user_input[CONF_AUTOMATIC_ADD],
+                CONF_EFFECT_SPEED: user_input[CONF_EFFECT_SPEED],
+            }
+
+            if CONF_CONFIGURE_DEVICE in user_input:
+                self._configure_device = user_input[CONF_CONFIGURE_DEVICE]
+                return await self.async_step_configure_device()
+
+            if CONF_REMOVE_DEVICE in user_input:
+                device_id = user_input[CONF_REMOVE_DEVICE]
+                config_data = copy.deepcopy(dict(self._config_entry.data))
+                del config_data[CONF_DEVICES][device_id]
+
+                self.hass.config_entries.async_update_entry(
+                    self._config_entry, data=config_data
+                )
+
+                async_dispatcher_send(
+                    self.hass, SIGNAL_REMOVE_DEVICE, {"device_id": device_id}
+                )
+
+                options_data = self._config_entry.options.copy()
+                if device_id in options_data:
+                    del options_data[device_id]
+                options_data["global"] = self._global_options
+
+                return self.async_create_entry(title="", data=options_data)
+
+            if CONF_HOST in user_input:
+                device_name = (
+                    user_input[CONF_NAME]
+                    if CONF_NAME in user_input
+                    else user_input[CONF_HOST]
+                )
+                device_id = user_input[CONF_HOST].replace(".", "_")
+                device_data = {
+                    CONF_HOST: user_input[CONF_HOST],
+                    CONF_NAME: device_name,
+                }
+                config_data = copy.deepcopy(dict(self._config_entry.data))
+                config_data[CONF_DEVICES][device_id] = device_data
+
+                self.hass.config_entries.async_update_entry(
+                    self._config_entry, data=config_data
+                )
+
+                async_dispatcher_send(
+                    self.hass, SIGNAL_ADD_DEVICE, {device_id: device_data}
+                )
+
+            options_data = self._config_entry.options.copy()
+            options_data["global"] = self._global_options
+            return self.async_create_entry(title="", data=options_data)
+
+        existing_devices = {}
+
+        for device_id, device in self._config_entry.data[CONF_DEVICES].items():
+            existing_devices[device_id] = device.get(CONF_NAME, device[CONF_HOST])
+
+        options = {
+            vol.Optional(
+                CONF_AUTOMATIC_ADD,
+                default=self._config_entry.options.get("global", {}).get(
+                    CONF_AUTOMATIC_ADD, self._config_entry.data[CONF_AUTOMATIC_ADD]
+                ),
```

Comment:
We don't need to store the options in two places, we can instead pass options when creating the config entry in `async_step_user`, see https://github.com/home-assistant/core/pull/49912